### DISCUSSION TITLE
MATSim

### DISCUSSION THREAD

**Discussion by pandanwaffles** [Universite-Gustave-Eiffel/NoiseModelling#439] [Q&A] [answered]
Hi all,

I would like to enquire on some confusion I faced when doing the MATSim tutorial from the Noisemodelling documentation.

1. What is the reason to Create the 0dB Source table in step 5?

2. In step 6: Calculate Noise Maps, why are the noise level in the LDAY_GEOM negative values?
![image](https://user-images.githubusercontent.com/90554682/160328879-01ace29c-233e-444c-8105-4815cc19ab42.png)

Below is the LDEN_GEOM I generated for my MATSim scenario
![image](https://user-images.githubusercontent.com/90554682/160329301-25d4d0fa-74ab-4be9-8ec4-5f7bcf080ea3.png)

The output I got from RESULT_GEOM was in negative values as well. Is this right? If so, how should I visualise this on QGIS to make it non negative?
![image](https://user-images.githubusercontent.com/90554682/160329162-501ed854-6421-4262-913e-d15acc7d8e0b.png)

Would really appreciate your help and answer! Thank you!

Best regards,

**Comment by Nitnelav** [comment, marked as answer]
Hi, yes so,

With MATSIM, you get traffic conditions (LV, LV_SPD,...) every 15min or 60min. To avoid computing the propagation for each traffic condition, we compute the attenuation between sources (roads) and receivers before applying it to every traffic condition.

The attenuation is computed by setting all roads Lw to 0dB (this is the SOURCE_0DB table) and running the Noise_Level_From_Source script with the confExportSourceId parameter.

The result (LDAY_GEOM table) then has all negative values as it represents the Lp-Lw attenuation between every source-receiver.

Then to get the Lp for  a specific traffic condition, we just have to arithmetically add the roads Lw in the LDAY_GEOM table and sum the contributing sources for every receiver.

Does it answer your question ?

**Reply by pandanwaffles** [reply]
Hi!

Thanks for the prompt reply!

Just to clarify, for the last part,

"Then to get the Lp for a specific traffic condition, we just have to arithmetically add the roads Lw in the LDAY_GEOM table and sum the contributing sources for every receiver."

Is this computed when generating RESULT_GEOM at Step 6 : Calculate Noise Maps using the Noise_From_Attenuation_Matrix WPS bloc? Or do we have to compute it manually?

Thank you so much for your help!

Best regards,

**Reply by Nitnelav** [reply]
Yes indeed, it's done in the following request in the Noise_From_Attenuation_Matrix script : 
```sql
CREATE TABLE RESULTS ( 
            PK integer PRIMARY KEY AUTO_INCREMENT,
            IDRECEIVER integer,
            THE_GEOM geometry,
            HZ63 double precision,
            HZ125 double precision,
            HZ250 double precision,
            HZ500 double precision,
            HZ1000 double precision,
            HZ2000 double precision,
            HZ4000 double precision,
            HZ8000 double precision,
            TIMESTRING varchar
        ) AS
        SELECT NULL, lg.IDRECEIVER,  lg.THE_GEOM,
            10 * LOG10( SUM(POWER(10,(mrs.LW63 + lg.HZ63) / 10))) AS HZ63,
            10 * LOG10( SUM(POWER(10,(mrs.LW125 + lg.HZ125) / 10))) AS HZ125,
            10 * LOG10( SUM(POWER(10,(mrs.LW250 + lg.HZ250) / 10))) AS HZ250,
            10 * LOG10( SUM(POWER(10,(mrs.LW500 + lg.HZ500) / 10))) AS HZ500,
            10 * LOG10( SUM(POWER(10,(mrs.LW1000 + lg.HZ1000) / 10))) AS HZ1000,
            10 * LOG10( SUM(POWER(10,(mrs.LW2000 + lg.HZ2000) / 10))) AS HZ2000,
            10 * LOG10( SUM(POWER(10,(mrs.LW4000 + lg.HZ4000) / 10))) AS HZ4000,
            10 * LOG10( SUM(POWER(10,(mrs.LW8000 + lg.HZ8000) / 10))) AS HZ8000,
            mrs.TIMESTRING AS TIMESTRING
        FROM LDAY_GEOM  lg 
        INNER JOIN   MATSIM_ROADS mr ON lg.IDSOURCE = mr.PK
        INNER JOIN MATSIM_ROADS_STATS mrs ON mr.LINK_ID = mrs.LINK_ID
        GROUP BY lg.IDRECEIVER, lg.THE_GEOM, mrs.TIMESTRING;
```

**Reply by pandanwaffles** [reply]
Noted! Thank you for clarifying. 

Below is the output of my RESULT_GEOM from Step 6 : Calculate Noise Maps using the Noise_From_Attenuation_Matrix WPS bloc

![image](https://user-images.githubusercontent.com/90554682/161223623-f40d81c3-8663-427d-91b1-a1c04f18d589.png)

The values I got was still negative. What could this mean? Is there an error with my output?

Your clarification is most appreciated! Thank you! 

Best regards,

**Reply by Nitnelav** [reply]
In RESULT_GEOM, assuming nothing went wrong, having negative values means : 

- there is no traffic on the roads arround the receiver at the given timestamp (it often happens at night when there is very little traffic)
- there is no roads close enough to the receiver (not close enough = not in the propagation radius = no source-receiver attenuation exists in LDAY_GEOM)

Can you share details on your project ? Are you just trying to reproduce the example or are you using another Matsim simulation ?

**Reply by Nitnelav** [reply]
Can you look into your MATSIM_ROADS_STATS and MATSIM_ROADS_TRAFFIC tables ? Is there any traffic at all ?

**Reply by pandanwaffles** [reply]
Hi! Thank you for your reply once again!
I'm just working on a school project and experimenting with MATSim and Noisemodelling. Hence, I created a small MATSim scenario with less than 5 inputs. 
Thank you for the details I have checked the 2 tables and I think I found the issue here but do correct me if my assumption is incorrect!

As my MATSim simulation was based on a really small scenario hence the traffic data was little.

Based on MATSIM_ROADS_STATS table only 2 traffic stats measurements
<img width="537" alt="MATSIM_ROADS_STATS data" src="https://user-images.githubusercontent.com/90554682/161372405-bd1b397b-b610-4701-a68c-84e8bbd76654.PNG">

& MATSIM_ROADS table only 3 traffic recorded
<img width="610" alt="MATSIM_ROADS data" src="https://user-images.githubusercontent.com/90554682/161372473-8cccfb7c-0d24-4a52-9c77-21889f486396.PNG">

Resulting in only 4 noise measurements in RESULTS_GEOM and the rest would be negative values.
![image](https://user-images.githubusercontent.com/90554682/161372530-2f5e1fce-a421-4ff9-b9e2-74a5e743ddbd.png)

Would my understanding/assumption be right? Thank you so much for clearing my doubt!

Best regards,

**Reply by Nitnelav** [reply]
Yep ! That looks coherent.
The rows in the MATSIM_ROADS do not correspond to "traffic recorded" but to "road segement" or "matsim link", the acutal traffic is in MATSIM_ROADS_TRAFFIC and the power levels are in MATSIM_ROADS_STATS. All are linked byt the `LINK_ID` field

**Reply by pandanwaffles** [reply]
I see, Noted with thanks!
Really appreciate the clarification and help! :)

Best regards,
